crates/k_means/src/approx.rs

```diff
@@ -0,0 +1,207 @@
+// This software is licensed under a dual license model:
+//
+// GNU Affero General Public License v3 (AGPLv3): You may use, modify, and
+// distribute this software under the terms of the AGPLv3.
+//
+// Elastic License v2 (ELv2): You may also use, modify, and distribute this
+// software under the Elastic License v2, which has specific restrictions.
+//
+// We welcome any commercial collaboration or support. For inquiries
+// regarding the licenses, please contact us at:
+// vectorchord-inquiry@tensorchord.ai
+//
+// Copyright (c) 2025 TensorChord Inc.
+
+use crate::square::{Square, SquareMut, partition_mut};
+use crate::{KMeans, This};
+use always_equal::AlwaysEqual;
+use distance::Distance;
+use rand::SeedableRng;
+use rand::rngs::StdRng;
+use rayon::iter::{IndexedParallelIterator, IntoParallelRefMutIterator, ParallelIterator};
+use std::collections::BinaryHeap;
+
+struct Approx<'a> {
```

Comment:
Please change all file names and variable names from `approx` to `hierarchical`.